From 364d467575b987fa27ad1967f13af4b35e304c70 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Wed, 19 Jun 2013 14:04:00 +0200
Subject: [PATCH 081/628] drm: rcar-du: Add default modes to VGA connector

According to the DRM developers drivers must not manually add default
modes to VGA connectors in an unknown connection state. Until a proper
solution is found (possibly by handling default modes in the DRM core,
as for connectors in the connected state), this hack is a convenience
that avoids forcing userspace to configure the modes manually.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/gpu/drm/rcar-du/rcar_du_vgacon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c b/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
index 9d7e5c9..a03bab1 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
@@ -23,7 +23,7 @@
 
 static int rcar_du_vga_connector_get_modes(struct drm_connector *connector)
 {
-	return 0;
+	return drm_add_modes_noedid(connector, 1280, 768);
 }
 
 static const struct drm_connector_helper_funcs connector_helper_funcs = {
-- 
2.7.4

